src/bin/pg_autoctl/file_utils.c

```diff
@@ -161,12 +188,9 @@ write_file(char *data, long fileSize, const char *filePath)
 bool
 append_to_file(char *data, long fileSize, const char *filePath)
 {
-	FILE *fileStream = NULL;
-
-	fileStream = fopen(filePath, "ab");
+	FILE *fileStream = fopen_with_umask(filePath, "ab", O_APPEND | O_CREAT, 0644);
 	if (fileStream == NULL)
 	{
-		log_error("Failed to open file \"%s\": %s", filePath, strerror(errno));
```

Comment:
Please add a comment about not logging on purpose, as we do everywhere in the code:

``` c
    /* errors have already been logged */
```